core/serial_ftdi.c

```diff
@@ -467,7 +467,7 @@ static dc_status_t serial_ftdi_write (void **userdata, const void *data, size_t
 
 		// Calculate the expected duration (microseconds). A 2 millisecond fudge
 		// factor is added because it improves the success rate significantly.
-		unsigned long expected = 1000000.0 * device->nbits / device->baudrate * size + 0.5 + 2000;
+		unsigned long expected = 1000000 * device->nbits / device->baudrate * size + 2000;
```

Comment:
I don't think it was by design, at least. If it avoided a overflow it was probably by accident.

Some quick googleing told me that the best source of android cpu market shares comes from unity game statistics, https://hwstats.unity3d.com/mobile/cpu-android.html , and that says 98.1% are ARMv7, so we need to support 32-bit there.

I'm going to take a closer look at this and try to figure out a sane solution. Meanwhile you can take the rest if you'd like. Or do you want me to rewrite the pr without this patch?